### PR TITLE
New C String Formatting Cheat Sheet

### PR DESCRIPTION
**Pull request by richiebful**
**What does your Instant Answer do?**

This instant answer provides a cheat sheet for printf string formatting in the C standard library.

**What is the data source for your Instant Answer? (Provide a link if possible)**

http://www.cplusplus.com/reference/cstdio/printf/

**Why did you choose this data source?**
It is a popular, comprehensive, and accurate source, which meets the existing standards for C programming.

**Are there any other alternative (better) data sources?**
There are other sources, such as the actual standard, but that costs money.

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
This Instant Answer will appeal to software developers.

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
No.

**Are you having any problems? Do you need our help with anything?**
No problems at all.

**Where did you hear about DuckDuckHack? (For first time contributors)**
I heard about it when DDG was doing its outreach campaign with t-shirts.

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
stringformattinggoodiescrn

-----
IA Page: https://duck.co/ia/view/string_fmt_cheat_sheet

### CODE REVIEW

**Comment by moollaza** [comment]
@richiebful thanks a lot for contributing, we really appreciate it! We'll be sure to review this shortly and give you some feedback!

**Comment by zachthompson** [comment]
@richiebful A few comments.  The two files submitted are different.  If you want to multiple triggers, the additional file names should be symlinks, e.g. ln -s string-formatting.json string-fmt.json.  So let's pick one of the files to keep and make the other a symlink.

The names of the files should probably contain "c-", e.g. "c-string-formatting.json", as these might not apply in other langs.  I would also consider adding aliases for "c++-", "printf", "sprintf", "snprintf", etc.

The order of sections would best be the same as in the format specifier.  The "length" section is missing as well.

**Comment by moollaza** [comment]
@richiebful Ping! I'm just checking in to see if you're still interested in finishing this one up? We'd love to see this go live!

There's absolutely no rush, just wanted to make sure you're not stuck on anything.

**Comment by zekiel** [comment]
https://duck.co/ia/view/string_fmt_cheat_sheet

**Comment by moollaza** [comment]
@richiebful Ping! I'm just checking in to see if you're still interested in finishing this one up? We'd love to see this go live!

There's absolutely no rush, just wanted to make sure you're not stuck on anything.

**Comment by moollaza** [comment]
@richiebful Ping! I'm just checking in again to see if you're still interested in finishing this one up? We'd love to see this go live!

BTW, thanks for doing such a great job so far! :cake: We really appreciate you taking the time to contribute. As a thanks, after this is merged we'd like to write a short blog post about you and your Instant Answer.

If you're interested please send an email to open@duckduckgo.com, with the subject "DuckDuckHack Dev Appreciation".

Please provide your name and include a link to this pull request as well.

We'll get in touch for more details!

**Comment by moollaza** [comment]
@richiebful I'm closing this one for now, but if you'd like to finish this up, we'll gladly open it back up for you!

File: share/goodie/cheat_sheets/string-formatting.json [hunk 1]
```diff
@@ -0,0 +1,142 @@
+{
```
Comment (moollaza) [inline share/goodie/cheat_sheets/string-formatting.json]: This file can be removed. We now have an `aliases` property that can be used instead:

```json
"name": "String Formatting",
"aliases": [ "string formatting" ],
...
```

**Pull request closed without merging**